src/script/util/ComponentUtil.ts

```diff
@@ -20,30 +20,62 @@
-  bindings?: never;
-  template: string;
-}
+type Subscribables<T> = {
+  [Key in keyof T]: T[Key] extends ko.Subscribable ? T[Key] : never;
+};
 
-interface RegisterReactComponentWithBindings<T> extends RegisterReactComponent<T> {
-  bindings: string;
-  template?: never;
-}
+type UnwrappedValues<T, S = Subscribables<T>> = {
+  [Key in keyof S]: Unwrapped<S[Key]>;
+};
+
+const resolveObservables = <C extends keyof Subscribables<P>, P extends Partial<Record<C, ko.Subscribable>>>(
+  object: P,
+  children?: C[],
+): UnwrappedValues<P> => {
+  const properties = children ?? (Object.keys(object).filter(key => key !== '$raw') as C[]);
+  return properties.reduce<UnwrappedValues<P>>((acc, child) => {
+    acc[child] = ko.unwrap(object?.[child]);
+    return acc;
+  }, {} as UnwrappedValues<P>);
+};
 
 /**
- * Registers a static react component against the ko world.
- * A Static component is a component that will not get updates for the observables directly given to it.
- * It can still react to nested observables given by using the `useKoSubribableChildren` hook.
- * A Static component is much optimal, in term of performance, that calling registerReactComponent (that will get a full state update everytime an observable changes).
+ * Will subscribe to all the observable properties of the object and call the onUpdate callback everytime one observable updates
+ *
+ * @param object The object containinig some observable properties
+ * @param onUpdate The callback called everytime an observable emits. It will only give back the slice of the object that have updates
+ * @param children? An optional list of properties to watch (by default will watch for all the observable properties)
+ */
+const subscribeProperties = <C extends keyof Subscribables<P>, P extends Partial<Record<C, ko.Subscribable>>>(
+  object: P,
+  onUpdate: (updates: Partial<UnwrappedValues<P>>) => void,
+  children?: C[],
+) => {
+  const properties = children ?? (Object.keys(object).filter(key => key !== '$raw') as C[]);
+  onUpdate(resolveObservables(object, children));
+
+  const subscriptions = properties
+    .filter(child => ko.isSubscribable(object?.[child]))
+    .map(child => {
+      const subscribable = object[child];
+      return subscribable?.subscribe((value: Unwrapped<typeof subscribable>) => {
+        onUpdate({[child]: value} as Partial<UnwrappedValues<P>>);
+      });
+    });
+
+  return {
+    dispose: () => subscriptions.forEach(subscription => subscription?.dispose()),
+  };
+};
+
+/**
+ * Registers a react component against the ko world.
  *
  * @param name Name of the component to register. can be used a `<component-name>` directly in ko
- * @param {component}
+ * @param component The React component to register
  */
-export function registerStaticReactComponent<Props>(name: string, component: React.ComponentType<Props>) {
+export function registerReactComponent<Props>(name: string, component: React.ComponentType<Props>) {
```

Comment:
This is the main change of this PR. 
It removes the `registerStaticReactComponent` and replaces it with a single optimized `registerReactComponent`. 

It both handled regular component (that do not have observables as direct input) and ko dynamic components (with observables as direct inputs).